algebird-test/src/main/scala/com/twitter/algebird/ApproximateProperty.scala

```diff
@@ -0,0 +1,137 @@
+package com.twitter.algebird
+
+import org.scalacheck.{ Gen, Prop }
+import org.scalacheck.util.Pretty
+
+trait ApproximateProperty {
+  type Exact
+  type Approx
+  type Input
+  type Result
+
+  def exactGenerator: Gen[Exact]
+  def inputGenerator(e: Exact): Gen[Input]
+
+  def makeApproximate(e: Exact): Approx
+
+  def exactResult(e: Exact, i: Input): Result
+  def approximateResult(a: Approx, i: Input): GeneralizedApproximate[Result]
+}
+
+trait GeneralizedApproximate[T] {
+  def withProb: Double
+  def contains(t: T): Boolean
```

Comment:
seems like what we really want is
```
trait ApproximateSet[S, -T] {
  def contains(set: S, t: T): ApproximateBoolean 
}
```
and use that as a typeclass rather than implicit conversions.
```
object ApproximateSet {
  def contains[S, T](s: S, t: T)(implicit as: ApproximateSet[S, T]): ApproximateBoolean = as.contains(s, t)
  implicit def fromApproximate[N: Numeric]: ApproximateSet[Approximate[N], N] = new ApproximateSet ...
}
```